pkg/envoy/config.go

```diff
@@ -1,9 +1,29 @@
 package envoy
 
-import k8s "github.com/openservicemesh/osm-health/pkg/kubernetes"
+import (
+	"fmt"
 
-// GetEnvoyConfig returns the Envoy config for the given pod.
-func GetEnvoyConfig(pod k8s.Pod) (*Config, error) {
+	v1 "k8s.io/api/core/v1"
+)
+
+// tempStruct implements ConfigGetter and is temporary until https://github.com/openservicemesh/osm-health/pull/14 is merged
+// TODO(draychev): delete this
+type tempStruct struct {
+	*v1.Pod
+}
+
+// GetConfig implements ConfigGetter
+func (mcg tempStruct) GetConfig() (*Config, error) {
+	return &Config{}, nil
+}
+
+// GetObjectName implements ConfigGetter
+func (mcg tempStruct) GetObjectName() string {
+	return fmt.Sprintf("%s/%s", mcg.Pod.Namespace, mcg.Pod.Name)
+}
+
+// GetEnvoyConfigGetterForPod returns a ConfigGetter struct, which can fetch the Envoy config for the given pod.
+func GetEnvoyConfigGetterForPod(pod *v1.Pod) (ConfigGetter, error) {
 	// TODO(draychev): Get the config from the Envoy sidecar of the given pod.
-	return nil, nil
+	return tempStruct{Pod: pod}, nil
 }
```

Comment:
@johnsonshi Once you have the code to fetch Envoy config -- we'll reuse that function here to fetch config for a given pod.

Reply:
We'll be using `openservicemesh/osm/pkg/cli`'s `GetEnvoyProxyConfig` for this.

Reply:
@johnsonshi Exactly right - here is where #19 is augmented w/ the new function: https://github.com/openservicemesh/osm-health/pull/25 (WIP)